src/ape/managers/compilers.py

```diff
@@ -27,11 +27,12 @@ class CompilerManager:
 
     config: ClassVar[ConfigManager] = injected_before_use()  # type: ignore
     plugin_manager: ClassVar[PluginManager] = injected_before_use()  # type: ignore
+    _registered_compilers_cache: Dict[Path, Dict[str, CompilerAPI]] = {}
 
     def __repr__(self):
         return f"<CompilerManager len(registered_compilers)={len(self.registered_compilers)}>"
 
-    @cached_property
+    @property
```

Comment:
Note: still cached, though now it is cached by project path. So if you change project paths, it will re-load.
This seems to work for what I need it for!